### PR TITLE
Fix RawGit Not Refreshing Resources on Master

### LINKED ISSUE #589: The Master Snapshot from the RawGit CDN will never refresh.
https://github.com/rgrove/rawgit/issues/41

We need instead of loading just the page from the master branch (which is cached one and NOT flushable as described in the link above) we need some kind of unique identifier to get the version of the file we actually want.

I propose adding the master branch's HEAD commit SHA in order to create the unique URL. This way changes merged to master will actually make the page update.

### PR DESCRIPTION
**Pull request by tn12787**
fixes #589.

This PR ensures that when a commit is added to `master` the RawGit CDN resources are refreshed, avoiding the CDN caching mechanism which caches a given URL once *permanently*.

### CODE REVIEW

**Comment by tn12787** [comment]
👍 (already in production).

n.b. this patch is working if this url has the new css: https://www.cyberbotics.com/doc/reference/radar

File: doc.php [hunk 1]
```diff
@@ -47,7 +59,8 @@
         'anchor':     window.location.hash.substring(1),
         'branch':     '$branch',
         'repository': '$repository',
-        'url':        'https://raw.githubusercontent.com/$repository/webots-doc/'
+        'url':        'https://raw.githubusercontent.com/$repository/webots-doc/',
+        'rawgiturl':  '$rawgiturl'
```
Comment (fabienrohrer) [inline doc.php]: If I'm correct, this seems to be unused.

Comment (tn12787) [reply doc.php]: it is unused, but I figured it would be useful for debugging purposes, no?

Comment (fabienrohrer) [reply doc.php]: I would say no: when we would like to debug, we have to introduce new variables etc., but introducing them in prevision to debug later is complexifying the code.

**Comment by tn12787** [comment]
👍

File: doc.php [hunk 2]
```diff
@@ -35,8 +35,20 @@
   }
   if (!isset($repository))
     $repository = 'omichel';
+
+  # get HEAD commit SHA, to ensure that when master is updated the latest version is cached by the CDN
```
Comment (fabienrohrer) [inline doc.php]: All this sequence is only used when `$branch === ''`, it would seem more logical to me to move it in the test below.

**Comment by stefaniapedrazzi** [comment]
Just note that uploading files on the FTP and put the in production before review is a bad habit.

**Comment by tn12787** [comment]
Sorry, it's an ugly habit, yes.

**Comment by tn12787** [comment]
GitHub has strict rate limiting, so I've added a fallback to use the dev version of rawgit at the **absolute worst**, in case GitHub refuses the request for the head SHA.

In general though, the same CDN-based mechanism will be used first. This way we can always load the current master branch snapshot, but with the consequence that it could be a little slower to load only if GitHub refuses our API request.

**Comment by tn12787** [comment]
👍

**Pull request merged**